src/meta/src/hummock/manager/compaction/compaction_group_schedule.rs

```diff
@@ -0,0 +1,316 @@
+use risingwave_hummock_sdk::CompactionGroupId;
+use risingwave_pb::hummock::compact_task::TaskStatus;
+use risingwave_pb::hummock::{PbGroupMerge, PbStateTableInfoDelta};
+use thiserror_ext::AsReport;
+
+use crate::hummock::error::{Error, Result};
+use crate::hummock::manager::transaction::HummockVersionTransaction;
+use crate::hummock::manager::{commit_multi_var, HummockManager};
+use crate::hummock::metrics_utils::remove_compaction_group_in_sst_stat;
+
+impl HummockManager {
+    /// Splits a compaction group into two. The new one will contain `table_ids`.
+    /// Returns the newly created compaction group id.
+    pub async fn split_compaction_group(
+        &self,
+        parent_group_id: CompactionGroupId,
+        table_ids: &[StateTableId],
+        partition_vnode_count: u32,
+    ) -> Result<CompactionGroupId> {
+        let result = self
+            .move_state_table_to_compaction_group(parent_group_id, table_ids, partition_vnode_count)
+            .await?;
+
+        Ok(result)
+    }
+
+    pub async fn merge_compaction_group(
+        &self,
+        group_1: CompactionGroupId,
+        group_2: CompactionGroupId,
+    ) -> Result<()> {
+        let compaction_guard = self.compaction.write().await;
+        let mut versioning_guard = self.versioning.write().await;
+        let versioning = versioning_guard.deref_mut();
+        // Validate parameters.
+        if !versioning.current_version.levels.contains_key(&group_1) {
+            return Err(Error::CompactionGroup(format!("invalid group {}", group_1)));
+        }
+
+        if !versioning.current_version.levels.contains_key(&group_2) {
+            return Err(Error::CompactionGroup(format!("invalid group {}", group_2)));
+        }
+
+        let state_table_info = versioning.current_version.state_table_info.clone();
+        let member_table_ids_1 = state_table_info
+            .compaction_group_member_table_ids(group_1)
+            .iter()
+            .cloned()
+            .collect_vec();
+
+        let member_table_ids_2 = state_table_info
+            .compaction_group_member_table_ids(group_2)
+            .iter()
+            .cloned()
+            .collect_vec();
+
+        let mut combine_1 = member_table_ids_1.clone();
+        combine_1.extend_from_slice(&member_table_ids_2);
+
+        let mut combine_2 = member_table_ids_2;
+        combine_2.extend_from_slice(&member_table_ids_1);
+
+        if !combine_1.is_sorted() && !combine_2.is_sorted() {
+            return Err(Error::CompactionGroup(format!(
+                "invalid merge group_1 {} group_2 {}",
+                group_1, group_2
+            )));
+        }
+
+        let mut left_group_id = group_1;
+        let mut right_group_id = group_2;
+        let combine_member_table_ids = if combine_1.is_sorted() {
+            combine_1
+        } else {
+            std::mem::swap(&mut left_group_id, &mut right_group_id);
+            combine_2
+        };
```

Comment:
nits: it is unnecessary to have three `is_sorted()` check here given that `compaction_group_member_table_ids` returns back a BTreeSet. We can make the logic simpler yet slightly more efficient:
```rust
        debug_assert!(!member_table_ids_1.is_empty());
        debug_assert!(!member_table_ids_2.is_empty());
        debug_assert!(member_table_ids_1.is_sorted());
        debug_assert!(member_table_ids_2.is_sorted());
        
        // Make sure `member_table_ids_1` is smaller than `member_table_ids_2`
        let (left_group_id, right_group_id) =
            if member_table_ids_1.first().unwrap() < member_table_ids_2.first().unwrap() {
                (group_1, group_2)
            } else {
                std::mem::swap(&mut member_table_ids_1, &mut member_table_ids_2);
                (group_2, group_1)
            };

        // We can only merge two groups with non-overlapping member table ids
        if member_table_ids_1.last().unwrap() >= member_table_ids_2.first().unwrap() {
            return Err(Error::CompactionGroup(format!(
                "invalid merge group_1 {} group_2 {}",
                left_group_id, right_group_id
            )));
        }

        let combined_memtable_table_ids =
            member_table_ids_1.iter().chain(member_table_ids_2.iter());

        debug_assert!(combined_memtable_table_ids.is_sorted());
```